decision-records/2024-11-07--dashboard-title-collision.md

```diff
@@ -0,0 +1,85 @@
+# Dashboard title collision in Grafana
+**Date:** 2024-11-07<br/>
+**Authors:** @lucabello @sed-i
+
+
+## Context and Problem Statement
+Dashboard json includes keys such as in the following example:
+```bash
+$ cat metrics-dashboard.json | jq '{"title", "uid", "version", "tags"}'
+{
+  "title": "Loki Operator Overview",
+  "uid": "loki",
+  "version": 1,
+  "tags": [
+    "loki"
+  ]
+}
+```
+
+Grafana [documented](https://grafana.com/docs/grafana/latest/administration/provisioning/#reusable-dashboard-urls) that dashboards should not have the same *title* within a folder, or the same *uid* within
+installations, which causes some problems in different scenarios.
+
+This ADR addresses **title** collisions.
+
+Dashboards from different charms may happen to have the same title:
+
+`loki(dashboard "Overview") <--> grafana <--> tempo(dashboard "Overview")`
+
+If both charms have a dashboard simply named "Overview", we'll have problems.
+We should not put charm authors in a position where they need to be mindful of an entire hypothetical deployment and try
+to avoid potential title collisions.
+
+
+## Requirements
+1. Dashboards must not have the same *title* within a folder.
+2. Juju admins, not charm authors, should have the final say on how dashboards are organized in sub-folders.
+
+## Decision
+(2/P).
```

Comment:
We came to a decision at this point in the document before reading the alternatives. I would move the Decision to to bottom of the document.

Reply:
Inlined the decision.
The section is at the top per template.